onyo/lib/commands.py

```diff
@@ -812,14 +818,15 @@ def onyo_set(inventory: Inventory,
 
         if ui.request_user_response("Update assets? (y/n) "):
             if not message:
-                operation_paths = [
-                    op.operands[0].get("path")
+                operation_paths = deduplicate([
+                    op.operands[0].get("path").relative_to(inventory.repo.git.root)
                     for op in inventory.operations
-                    if op.operator == OPERATIONS_MAPPING['modify_assets']]
+                    if op.operator == OPERATIONS_MAPPING['modify_assets']])
                 message = inventory.repo.generate_commit_message(
-                    cmd="set",
-                    keys=[str(k) for k in keys.keys()],
-                    modified=operation_paths)
+                    format_string="set [{len}] ({keys}): {operation_paths}",
+                    len=len(operation_paths),
+                    keys=','.join([str(k) for k in keys.keys()]),
```

Comment:
This already is a list of strings.

```suggestion
                    keys=','.join(keys.keys()),
```

Reply:
Moreover, this confuses me a bit. What is the special treatment of `list` in `generate_commit_message` for, when here you do the `', '.join(...` outside?